### PR TITLE
Only populate real_value_cache during export

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #90468

Fixes https://github.com/pytorch/torchdynamo/issues/1950

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @Guobing-Chen @chunyuan-w @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### CODE REVIEW

**Comment by voznesenskym** [comment]
Fails tests

File: torch/_dynamo/output_graph.py [hunk 1]
```diff
@@ -329,7 +329,7 @@ def register_attr_or_module(
 
             def wrap_name(module_key):
                 return wrap_fx_proxy(
-                    self,
+                    self.root_tx,
```
Comment (ezyang) [inline torch/_dynamo/output_graph.py:332]: I think this was a bug

**Review by voznesenskym — APPROVED** [review]

**Comment by ezyang** [comment]
unfortunately dynamo failures probably real?

**Comment by voznesenskym** [comment]
> unfortunately dynamo failures probably real?

Want me to commandeer to get tests passing?

**Comment by ezyang** [comment]
Yes please

File: test/test_masked.py [hunk 2]
```diff
@@ -313,6 +313,7 @@ def test_mask_layout(self, layout, device, dtype, op, sample_inputs):
             expected = op.op(r_inp, *r_args, **r_kwargs)
             self.assertEqualMasked(actual, expected, outmask)
 
+    @skipIfTorchDynamo("Test does not work with TorchDynamo")
```
Comment (ezyang) [inline test/test_masked.py:316]: If you file a tracking issue it's nice to link to it for context for peeps in the future

Comment (voznesenskym) [reply test/test_masked.py:316]: yes

**Pull request closed without merging**